### ISSUE TITLE
Authors with duplicate email addresses get mixed up

### ISSUE THREAD

**Issue by grawity** [jonas/tig#424]
See for example https://github.com/atheme/charybdis, which (as an artifact of svn2git import) has several authors with the same `devnull@localhost` address:

```
androsyn <devnull@localhost>
jilles <devnull@localhost>
nenolod <devnull@localhost>
...
```

In tig's main (commit log) view, the name column only shows "androsyn" next to all such commits, even though the individual commits are shown as authored by "nenolod &lt;devnull@localhost&gt;" or such.

(If I enable `set mailmap`, it works fine, since the mailmap translates all such commits to unique emails.)

tig v1 not affected, only tig v2.x.

**Comment by jonas** [comment]
As you've found, Tig uses the email as the key in the author hashmap, where
the author name is taken from the first occurrence. Given that `set
mailmap` works, I guess this is low priority?

On Sun, Jul 12, 2015 at 3:36 PM, Mantas Mikulėnas <notifications@github.com>
wrote:

> See for example the atheme/charybdis repo, which (as an artifact of
> svn2git import) has several authors with the same devnull@localhost
> address:
>
> androsyn <devnull@localhost>
> jilles <devnull@localhost>
> nenolod <devnull@localhost>
> ...
>
> In tig's main (commit log) view, all commits have the name "androsyn" next
> to them, even though the individual commits are shown as authored by
> "nenolod <devnull@localhost>".
>
> (If I enable set mailmap, it works fine, since the mailmap translates all
> such commits to unique emails.)
>
> tig v1 not affected, only tig v2.x.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/jonas/tig/issues/424>.
>



-- 
Jonas Fonseca

**Comment by alangpierce** [comment]
I'm running into this as well. I wrote a [tool](https://github.com/alangpierce/bulk-decaffeinate) that makes a bunch of automated code changes, and I intentionally use a different author (but the same email address) when generating commits for those changes. For example, my normal author is `Alan Pierce <alan@benchling.com>`, but when I run the script, it uses `decaffeinate <alan@benchling.com>` as the author for the auto-generated commits. That way, someone using git blame will see the name as "decaffeinate" for those lines and know that they were generated by an automated tool and I don't necessarily have context on the code. But anyone can still look at the email field to see who ran the tool.

Now, people on my team who use tig are running the tool, and when they open tig, they see their name change to "decaffeinate" for *all* of their commits. It looks like every other git tool I know of handles this case correctly.

**Comment by jonas** [comment]
Fixed in #547
